### ISSUE TITLE
Migration from amazon-echo-ha-bridge

### ISSUE THREAD

**Issue by dbranagh** [bwssytems/ha-bridge#567]
Hi,
I am trying from amazon-echo-ha-bridge to migrate to the latest ha-bridge, but can't quite work out what needs configured and how.
How/where/what do I populate to duplicate the 'on url' of:
"http://localhost:8080/CMD?Sonoff4=ON"

(happens to be a call into openHab which then goes to mosquito which then goes to a sonoff switch - but I don't think that really matters)

Thanks,
David

**Comment by bwssytems** [comment]
Reading is your friend: https://github.com/bwssytems/ha-bridge/blob/master/README.md

**Comment by dbranagh** [comment]
Reading is fine. And i appreciate your work on github. Unfortunately my understanding from what i read does not answer my question. That is evidently my fault.
Thanks.

On 24 Mar 2017 8:30 pm, BWS Systems <notifications@github.com> wrote:

Reading is your friend: https://github.com/bwssytems/ha-bridge/blob/master/README.md

**Comment by ubergeeks** [comment]
Unfortunately I have been pulling my hair out also with the new UI and how to fix all of the items previously working fine with openhab. I have googled my way around for every comment and cannot find much of anything to help, very little openhab feedback for configurating the proper POST arrangements in this new UI and I have everything working fine via postman and browser urls, just cannot make it work in these new versions of ha-bridge. Need some HELP please.

**Comment by bwssytems** [comment]
So, in the UI, Type is for the communication selection. in this case it is HTTP device. Target Item is where your URL goes i.e. http://192.168.1.1/dosomething. HTTP Verb is the PUT/POST/GET. HTTP Body is for the payload that goes with the PUT/POST if there is one and then there is the Headers if you have something special and then the context of text/xml/json or others.

Then hit the Add button to the right of that line. then name your device and save it.

As far as openhab specifics, you would need to refer to their API guide.

**Comment by dbranagh** [comment]
Thanks for the response.
I select on the emulator webUI home->Add/edit
On this page I then set Device type as 'switch', Map Type as 'HTTP device', and in the 'On items' section I fill in again the type as 'HTTP device', the HTTP verb as 'GET', and the context type as 'text/html' (see screenshot below).

**Where should I put the URL I want it to get??** (the  "http://localhost:8080/CMD?Sonoff4=ON" quoted above)

Clicking 'Add' in the 'On items' section has no effect.
If I them click on the 'Add Bridge Device' at the top the device is added to the list of bridge devices.
If I try and test this device I get the error: "Request error, Pleae look in your habridge log: error ...... could not find url...."
The terminal o/p is: 2017-03-28 23:07:05,073 [qtp1235577049-18] WARN  com.bwssystems.HABridge.hue.HueMulator - Could not find url: 5 for hue state change request: thehabridgeuser from 192.168.1.125 body: {"on":true}


![hue1](https://cloud.githubusercontent.com/assets/26611251/24429813/9bf2a7da-140b-11e7-9352-5a0218c10d4c.jpg)

**Comment by dbranagh** [comment]
To partially answer my own question - if I put "http://localhost:8080/CMD?Sonoff4=ON" in the 'Target item' text box, then the 'Add' button beside will work. I can do a similar thing for the 'Off Items' and then "Update Bridge device".
The 'Test ON' and 'Test OFF' buttons now work.
However if I try to edit this device again the 'On Items' and 'Off Items' are once again blank - may be this is a bug.
Also I notice that the phrase/label "target item" does not appear in the readme: https://github.com/bwssytems/ha-bridge/blob/master/README.md

**Comment by bwssytems** [comment]
you need to hit save device at the top

**Comment by ubergeeks** [comment]
I have tried to find a working configuration for my previously working openhab itesm wit no joy. I had read the wikis and have re-read hat you suggested above . No joy.

To top it off it sounds like everytime i make a change I might not be hitting the correct button? Do i have to click add everytime and enter in a new line when i make a device edit or do i hit update bridge device at top or another button?

I also am not sure of the correct syntacx for things like body, etc for openhab items. Do they need to be in braces or brackets? I do not see where this is clear.

If someone could just post a sample working openhab rest example it would probably help alot of people out there trying to make this new version work. I am by no means a novice with the Openhab 1.x going a few years back and running 2.0 now. I have used the echobridge in early flavors with no issue and even your fork fine up to the UI redesign. I'm just trying to get it working for openhab. I have harmony working etc.

Thanks again for your work and efforts.

**Comment by dbranagh** [comment]
Hi ubergeeks,
For the test device I have working my "device.db" entry is as follows:

`{"id":"5","uniqueid":"00:17:88:5E:D3:05-05","name":"Shelf light","mapId":"undefined-copy","mapType":"httpDevice","deviceType":"switch","offUrl":"[{\"item\":\"http://localhost:8080/CMD?Sonoff4\u003dOFF\",\"type\":\"httpDevice\",\"httpVerb\":\"GET\",\"contentType\":\"text/html\"}]","onUrl":"[{\"item\":\"http://localhost:8080/CMD?Sonoff4\u003dON\",\"type\":\"httpDevice\",\"httpVerb\":\"GET\",\"contentType\":\"text/html\"}]","httpVerb":"GET","contentType":"text/html","contentBody":"http://localhost:8080/CMD?Sonoff4\u003dON","contentBodyOff":"http://localhost:8080/CMD?Sonoff4\u003dOFF","inactive":false,"noState":false,"offState":false,"description":"shelf light"}`

Note, if I try to edit this device through the 'Edit/copy' button the form appears blank.

Perhaps we will not need to use the bridge when the openhab Alexa skill is available - but I have no idea when this will be.

**Comment by dbranagh** [comment]
Then again maybe I missed the announcement by 1 month:  https://community.openhab.org/t/official-alexa-smart-home-skill-for-openhab-2/23533  
(although looks like OH2 required)

**Comment by ubergeeks** [comment]
Yes, but that uses the cloud service, prefer all  local but am also looking into it. I did migrate to OH2 a while back but still working with items statically defined in items files.

I just wanted to also report back I looked at my overall install and figured it must be related to the docker. The docker itself is fine, I was using a pipework docker to assign a different IP than the docker host, which for some reason is causing the communication to fail. I have pipework serving IP's to other dockers without issue, but there is something I'm missing. Anyway, set to use host IP now and all is well.

Great work and thanks again to BWS for the contribution and development of this bridge.
